File: TP01/TP01Q04 - Alteração Aleatória em Java/AlteracaoAleatoria.java
//TP01Q04 - Alteração aleatória em Java
//@author: Vitória Símil Araújo

import java.util.Random;

public class AlteracaoAleatoria {
    private static boolean isFim(String input) {
        return (input.charAt(0) == 'F' && input.charAt(1) == 'I' && input.charAt(2) == 'M');
    }

    public static String execute(String entrie, Random generator) {
        //get two random characters
        char first = ((char)('a' + (Math.abs ( generator.nextInt ( ) ) % 26 ) ));
        char second = ((char)('a' + (Math.abs ( generator.nextInt ( ) ) % 26 ) ));

        return execute(entrie, first, second);
    }

    private static String execute(String entrie, char first, char second) {
        //data
        String changedString = "";

        for ( char c : entrie.toCharArray()) {
            if ( c == first ) 
                changedString += second;
            else 
                changedString += c;
        } 

        return changedString;
    }

    public static void main(String[] args) {
        //data
        String entrie = "";  

        //create generator
        Random generator = new Random();
        generator.setSeed(4);

        //read entries archive
        while(!isFim(entrie = MyIO.readLine())) {
            MyIO.println((execute(entrie,generator)));
        }
    }
}

File: TP01/TP01Q16 - Alteração Aleatória em C/AlteracaoAleatoria.c
//TP01Q16 - Alteração Aleatória em C
//@author: Vitória Símil Araújo

#include <stdlib.h> 
#include <stdio.h> 
#include <time.h> 
#include <string.h>
#include <stdbool.h>

//functions
bool isEnd(char entrie[]);
char* execute(char entrie[]);
char* changeString(char entrie[], char first, char second);


int main(void) {
    //data
    char entrie[1000] = {};

    //create generator
    srand(4);

    //read the input archive
    scanf("%[^\n]", entrie);
    getchar();
    while ( !isEnd(entrie) ) {
        printf("%s\n", execute(entrie));
        scanf("%[^\n]", entrie);
        getchar();
    }
}

//method to verify if a string equals FIM
bool isEnd(char entrie[]) {
    return (entrie[0] == 'F' && entrie[1] == 'I' && entrie[2] == 'M');
}

char* execute(char entrie[]) {
    //get two random characters
    char first = (char) 'a' + rand() % 26;
    char second = (char) 'a' + rand() % 26;

    return changeString(entrie, first, second);
}

char* changeString(char entrie[], char first, char second) {    
    //data
    int size = strlen(entrie);

    for(int i = 0; i < size; i++ ) {
        if ( entrie[i] == first ) {
            entrie[i] = second;
        }
    }
    return entrie;
}